pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -80,13 +89,15 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.Tekton
 	if err := r.extension.Finalize(ctx, original); err != nil {
 		logger.Error("Failed to finalize platform resources", err)
 	}
-	logger.Info("Deleting cluster-scoped resources")
-	manifest, err := r.installed(ctx, original)
-	if err != nil {
-		logger.Error("Unable to fetch installed manifest; no cluster-scoped resources will be finalized", err)
-		return nil
+
+	if reflect.DeepEqual(lastAppliedManifest, mf.Manifest{}) {
+		lastAppliedManifest = r.manifest
+		if err := r.transform(ctx, &lastAppliedManifest, original); err != nil {
+			return err
+		}
 	}
```

Comment:
```suggestion
     lastAppliedManifest = r.manifest
     if err := r.transform(ctx, &lastAppliedManifest, original); err != nil {
	return err
    }
```

can we assume that when the finzlizer is called there will be no change in spec after the `DeletionTimeStamp` is set.
In that case, we can perform the deletion on transformed list of r.manifest.

Reply:
In short, any restart of controller pod will reset `lastAppliedManifest` to empty.